Convert the Trusted Clusters guide to a tutorial (and edit for different scopes)

ptgott opened the pull request:
See #10633

- Misc style/grammar/clarity tweaks
- Turn the Teleport Node Tunneling Admonition into a Details
  box so it can be invisible for Cloud users. In Cloud, Nodes
  must connect via Node Tunneling.
- Use Tabs components to add Cloud versions of CLI commands
- Only show the static join token method for self-hosted users
  via Tabs
- Use a Details box to show content relevant only for Enterprise
  and Cloud users
- Remove an Admonition that was duplicated in the Troubleshooting
  section

ptgott committed:
Edit the Trusted Clusters guide for Cloud

See #10633

- Misc style/grammar/clarity tweaks
- Turn the Teleport Node Tunneling Admonition into a Details
  box so it can be invisible for Cloud users. In Cloud, Nodes
  must connect via Node Tunneling.
- Use Tabs components to add Cloud versions of CLI commands
- Only show the static join token method for self-hosted users
  via Tabs
- Use a Details box to show content relevant only for Enterprise
  and Cloud users
- Remove an Admonition that was duplicated in the Troubleshooting
  section

ulysseskan commented on docs/pages/setup/admin/trustedclusters.mdx:
```diff
@@ -71,51 +72,96 @@ $ tsh ssh --cluster=leaf host
 ![Tunnels](../../../img/tunnel.svg)
 
-This setup works as follows:
+This setup works as follows. The "leaf" creates an outbound reverse SSH tunnel
```
Per https://english.stackexchange.com/a/394875 , I think you still need a colon after "as follows".

ulysseskan commented on docs/pages/setup/admin/trustedclusters.mdx:
```diff
@@ -71,51 +72,96 @@ $ tsh ssh --cluster=leaf host
-### Static Join Tokens
+### Static join tokens
 
 To create a static join token, update the configuration file on "root" cluster
```
Should say on **the** "root" cluster.

ulysseskan replied:
Although I'm not sure static join token section is applicable to cloud (yet), as customers cannot edit cloud's teleport.yaml , right?

ptgott replied:
I've aded a `Tabs` component with the `scope` property set around this H3 and the "Dynamic Join Tokens" H3, so these sections only appear if the reader has selected "OpenSource" or "Enterprise" in the dropdown menu.

ulysseskan commented on docs/pages/setup/admin/trustedclusters.mdx:
```diff
@@ -160,19 +204,70 @@ $ tctl tokens ls
+Users of Teleport will recognize that this is the same way you would add any
+Node to a cluster. 
+
 Now, the administrator of "leaf" must create the following resource file:
```
of **the** leaf

ptgott replied:
I think skipping the article is okay here since "leaf" is the name of the cluster, but it does sound a bit odd. I've suggested using "the leaf cluster" instead.

ulysseskan commented on docs/pages/setup/admin/trustedclusters.mdx:
````diff
@@ -277,11 +371,33 @@ spec:
+</TabItem>
+<TabItem scope={["cloud"]} label="Teleport Cloud">
+```yaml
+# Save this as root-cluster.yaml on the auth server of "leaf" and then execute:
````
This whole section seems technically correct but when viewing the file in github, is not formatted properly like the section above it.  Instead, what is supposed to look like yaml is jumbled together as a paragraph without line breaks in the proper places.

ptgott replied:
Does this look okay if you view the docs site locally with these steps?
- Open a clone of `gravitational/next`
- Run `git submodule update --init --recursive`
- Navigate to `content/9.0`
- Check out my branch
- Run `yarn dev`
- Open the page

ulysseskan replied:
Yes, yes it does.

ulysseskan commented on docs/pages/setup/admin/trustedclusters.mdx:
```diff
@@ -572,9 +700,29 @@ how your network security groups are configured on AWS.
+  questions above. 
+</TabItem>
+<TabItem scope={["cloud"]} label="Teleport Cloud">
+Troubleshooting "access denied" messages can be challenging. A Teleport administrator
```
This section is repeated and also appears directly above.  Also, the troubleshooting section as a whole seems mostly relevant to non-cloud installations.

ptgott replied:
There are two separate `TabItem` components here, one which is automatically selected for readers who have chosen "OpenSource" or "Enterprise" in the dropdown menu at the top of the docs site ("Self-Hosted"), and one for readers who have selected "Cloud". Are the correct instructions in each `TabItem`?

ulysseskan replied:
Yes, when viewing in yarn dev, it looks fine.

ptgott commented on docs/pages/includes/tctl.mdx:
```diff
@@ -1,22 +1,36 @@
 
-Verify that your Teleport client is connected:
+Verify that your Teleport client is connected by running the following on your
+Auth Service host:
```
This uses a `Details` box to show this advice if the reader has the "Enterprise" or "OpenSource" options selected in the dropdown menu at the top of the docs site, and shows the `Details` box below if the reader has "Cloud" selected. The `Details` box changes its visibility based on the `scope` and `scopeOnly` attributes.

ptgott committed:
Respond to PR feedback

ulysseskan reviewed (approved):
Looks great!

zmb3 commented on docs/pages/setup/admin/trustedclusters.mdx:
```diff
@@ -4,58 +4,59 @@ description: How to configure access and trust between two SSH and Kubernetes en
+<Details title="Teleport Node Tunneling" scope={["enterprise", "oss"]} scopeOnly opened>
 
-## Introduction
+ If you have a large number of devices on different networks, such as managed
```
I'm not sure the criteria for using tunneling is "a large number of devices."

Technically, node tunneling consumes **more** resources than direct dial, so your cluster would better handle a large number of devices if you did not use tunneling.

I would recommend we reword this so that networking is the main driver for tunneling. 
Something like:

If your nodes are deployed behind a firewall or otherwise not reachable by the Teleport Proxy Service, you can connect your nodes via Teleport Node Tunneling. Instead of connection to the Auth Service directly, ...

zmb3 commented on docs/pages/setup/admin/trustedclusters.mdx:
```diff
@@ -4,58 +4,59 @@ description: How to configure access and trust between two SSH and Kubernetes en
+## How Trusted Clusters work
+
+Teleport can partition compute infrastructure into multiple clusters. A cluster
+is a group of Teleport SSH Nodes connected to the cluster's Auth Service, which
```
```suggestion
is a group of Teleport resources connected to the cluster's Auth Service, which
```

Not sure of term (resources, agents, etc), but let's be clear that a cluster is more than SSH nodes and an Auth Service.

zmb3 commented on docs/pages/setup/admin/trustedclusters.mdx:
```diff
@@ -501,52 +623,59 @@ node_labels:
+- Checks that the certificate signature matches one of the Trusted Clusters.
 - Tries to find a local role that maps to the list of principals found in the certificate.
-- Checks if the local role allows the requested identity (UNIX login) to have access.
+- Checks if the local role allows the requested identity (Unix login) to have access.
```
Nit: in the PRs I've reviewed today, we've used all of these interchangeably:

- Unix login
- OS login
- Host login

Should we standardize on a term here?

ptgott replied:
I'm not sure we need to have a standard for this unless this is a Teleport-specific technical term. We use `logins` in user/role resource definitions, but the docs also sometimes use the term outside of that context. I'll add it to my list of possible topics to include in a style guide.

ptgott committed:
Address PR feedback

ptgott commented:
Since I need to audit this guide to complete #11841 and this PR has only received one review, I am going to use this branch to make my changes.

ptgott committed:
Turn the Trusted Clusters guide into a tutorial

See: #11841

The Trusted Clusters guide is organized as a conceptual introduction,
with configuration/command snippets used as illustrations. To make this
guide easier to follow, I have structured it as a step-by-step tutorial
where a user should be able to copy each command/config snippet on
their own environment, establish trust between clusters, and connect to
a remote Node.

Some more specific changes:

- Remove Details box re: Node Tunneling: This isn't strictly relevant
  to Trusted Clusters, so removing it shortens and simplifies what is
  quite a long guide.

- Make "How Trusted Clusters work" more concise and add the information
  to the introduction.

- Move long explanatory passages into Details boxes. Eventually, it
  would be great to split this guide into multiple guides that explain
  different topics in more depth (e.g., a section of the docs devoted
  to Trusted Clusters). For now, this is the quickest way to organize
  conceptual information without detracting from the tutorial structure.
